src/OmniSharp.Roslyn/OmniSharpWorkspace.cs

```diff
@@ -194,7 +194,7 @@ private void LogDeletionException(Exception e, string filePath)
         protected override void ApplyDocumentAdded(DocumentInfo info, SourceText text)
         {
             var project = this.CurrentSolution.GetProject(info.Id.ProjectId);
-            var fullPath = info.FilePath;
+            var fullPath = Path.GetFullPath(info.FilePath);
```

Comment:
When I was trying to make host.Workspace.TryApplyChanges(newSolution) work, it was sometimes throwing exceptions here since info.FilePath was directly in project folder (no directory), I kept this since it looks like a reasonable change. Also, is this method used only in tests or am I missing something?

Reply:
This is called on the roslyn side, it's possible that this may some weird problems.  It might make more sense for the test to be changed, so that it has a proper path (and even make sure the file exists)

Reply:
Still not sure about this, if you could try taking it out once the test has been fixed, and let us know what the errors your seeing are, it's probably something that needs to be fixed.

Reply:
This is an exception without this change:

```
OmniSharp.Tests.UpdateBufferFilterFacts.UpdateBuffer_TransientDocumentsDisappearWhenProjectAddsThem [FAIL]
      System.ArgumentException : Path cannot be the empty string or all whitespace.
      Stack Trace:
           at System.IO.Directory.CreateDirectory(String path)
           at OmniSharp.OmniSharpWorkspace.SaveDocumentText(DocumentId id, String fullPath, SourceText newText, Encoding encoding)
```

this is the code it causes it:
```
var dir = Path.GetDirectoryName(fullPath);
if (!Directory.Exists(dir))
{
    Directory.CreateDirectory(dir);
}
```
without Path.GetFullPath, this would also work:
```
var dir = Path.GetDirectoryName(fullPath);
if (!string.IsNullOrEmpty(dir) && !Directory.Exists(dir))
{
    Directory.CreateDirectory(dir);
}
```

Changing this specific test to use full path would fix the issue, but then all the other tests use filenames like "foo.cs" "bar.cs" etc. If DocumentInfo.FileName is allowed to be relative path in general, some check should be in order. Documentation is vague: us/dotnet/api/microsoft.codeanalysis.documentinfo?view=roslyn-dotnet.